Sources/Datadog/Datadog.swift

```diff
@@ -155,7 +155,16 @@ public class Datadog {
                 networkConnectionInfoProvider: networkConnectionInfoProvider,
                 carrierInfoProvider: carrierInfoProvider
             )
-            tracingAutoInstrumentation = TracingAutoInstrumentation(tracedHosts: configuration.tracedHosts)
+            if !configuration.tracedHosts.isEmpty {
+                let urlFilter = URLFilter(
+                    includedHosts: configuration.tracedHosts,
+                    excludedURLs: [
+                        configuration.logsEndpoint.url,
+                        configuration.tracesEndpoint.url
+                    ]
+                )
```

Comment:
i'm not sure if `URLFilter` should know `Datadog.Config` 🤔 
i'm thinking of a convenience init for `TracingAutoInstrumentation.init?(configuration:)`, the only place where `TracingAutoInst` knows how to use `Datadog.Config`

i made some changes, looks good to me, please have a look

Reply:
> i'm not sure if `URLFilter` should know `Datadog.Config` 🤔

If it's generic `URLFilter`, it shouldn't, but as we don't have need for reusing it (and even exposing a configuratble collection of excluded URLs), this could be simply `TracedURLsFilter(configuration: configuration)`.

> i made some changes, looks good to me, please have a look

Looks good 👌.

Reply:
hmm now i see 👌 yep we could do that as well 👍 
but i kept it generic in case we might re-use it for RUM or somewhere else